From 9fd064585780e795a25297f1ab8e64a97d6ad11e Mon Sep 17 00:00:00 2001
From: "Bandi, Kushal" <kushal.bandi@intel.com>
Date: Tue, 28 Aug 2018 19:14:02 -0700
Subject: [PATCH 2077/2367] media: intel-ipu4: psys: Adding missing compact32
 member.

There was a member defined in header which was missing
in the driver at time of mapping from userspace.
With this fix 32bit compatibility error gets resolved.

Change-Id: Ibef3ba68fe63e22202dc1f231dec90e4215a9256
Signed-off-by: Bandi, Kushal <kushal.bandi@intel.com>
---
 drivers/media/pci/intel/ipu-psys-compat32.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/pci/intel/ipu-psys-compat32.c b/drivers/media/pci/intel/ipu-psys-compat32.c
index 452fd38..df1bd4f 100644
--- a/drivers/media/pci/intel/ipu-psys-compat32.c
+++ b/drivers/media/pci/intel/ipu-psys-compat32.c
@@ -42,6 +42,7 @@ struct ipu_psys_command32 {
 	u32 pg_manifest_size;
 	u32 bufcount;
 	u32 min_psys_freq;
+	u32 frame_counter;
 	u32 reserved[2];
 } __packed;
 
@@ -69,6 +70,7 @@ get_ipu_psys_command32(struct ipu_psys_command *kp,
 	    get_user(kp->pg_manifest_size, &up->pg_manifest_size) ||
 	    get_user(kp->bufcount, &up->bufcount) ||
 	    get_user(kp->min_psys_freq, &up->min_psys_freq)
+		|| get_user(kp->frame_counter, &up->frame_counter)
 	    )
 		return -EFAULT;
 
-- 
2.7.4

